index.rst

```diff
@@ -58,24 +58,216 @@ We describe the design and implementation of the LSST Alert Distribution System,
+that can be used for real-time and continuous data pipelines.
+Kafka is a scalable pub/sub message queue based on a commit log.
+It is used in production at scale at companies such as LinkedIn,
+Netflix, and Microsoft to process over 1 trillion messages per day.
+
+Kafka collects messages from processes called "producers,"
+which are organized into distinct streams called "topics."
+Downstream "consumers" pull messages by subscribing to topics.
+Topics can be split into "partitions" that may be distributed
+across multiple machines and allow consumers to read in
+parallel as "consumer groups."
+Data can be replicated by deploying Kafka in cluster mode over several
+servers called "brokers."
+We will refer to these brokers below as "Kafka brokers" to distinguish
+from the LSST alert downstream "community brokers."
+
+For LSST alert distribution, Kafka and the accompanying Zookeeper
+can be deployed as Docker containers from the DockerHub image repository
+maintained by Confluent Inc., the team that created Kafka.
+The latest release of ``alert_stream`` uses Kafka and Zookeeper from
+Confluent platform release 4.1.1.
+The producer used for generating and sending data to Kafka and
+template scripts for consumers of the stream are provided in the GitHub
+repository at https://github.com/lsst-dm/alert_stream,
+which can also be built as a Docker image and deployed as containers.
 
 Alert Filtering
 ================
 
+Alert filters can be written using simple Python functions that act
+solely on the contents of the alert packets.
+Each filter can run separately and isolated in its own container.
+A filter is constructed as a consumer of the Kafka topic containing the
+full stream and a producer back to a Kafka topic for the filtered stream.
+This filtered stream can then be read by another consumer or directed
+to another output method for storage.
+
+Currently, filters are defined in ``lsst.alert.stream.filters`` in the
+alert_stream repository.
+Each filter is constructed as a class with a filter function.
+Below is an example of code defining a simple filter.
+
+.. code-block:: Python
+
+  class Filter001(AlertFilter):
+      def filter(self, alert):
+          if ((alert['diaSource']['snr'] > 5) &
+              (alert['diaSource']['diffFlux'] > 0.00003631)):  # 20th mag
+              return True
+          else:
+              return False
+
+Filters inherit from an AlertFilter base class.
+When the filter class is called on each alert, the visit ID is read.
+Up to 20 alerts per visit that pass the filter, i.e., return True, are
+forwarded to a topic for the filtered alerts, named for the class.
+In the above example, alerts are sent to a new topic named "Filter001."
+The filter detects the beginning of a new visit when the visit ID
+changes from one alert to the next.
+This assumes that all alerts from a visit will be received before
+the next visit's alerts arrive.
+Otherwise, more than 20 alerts may pass through the filter.
```

Comment:
I'd make clearer that nothing blows up if more than 20 alerts pass through; it's a coarse load-balancing measure.